projects/lights_control/inc/lights_signal_fsm.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+// This module handles state transitions for signal lights. There are 6 states in total.
+
+#include "event_queue.h"
+#include "fsm.h"
+#include "status.h"
+
+#include "lights_blinker.h"
+#include "lights_events.h"
+
+// Instance of lights signal.
+typedef struct LightsSignalFsm {
+  LightsBlinker blinker;
+  FSM fsm;
+} LightsSignalFsm;
+
+// Initializes state machines, and blinker.
+StatusCode lights_signal_fsm_init(LightsSignalFsm *lights_signal_fsm,
+                                  LightsBlinkerDuration blinker_duration);
+
+// Processes signal events.
+StatusCode lights_signal_fsm_process_event(LightsSignalFsm *lights_signal_fsm, const Event *event);
```

Comment:
You know if you exposed the FSM instead of wrapping it, you could just have an array of FSMs that you call `fsm_process_event` on in your main loop.

Reply:
I only have one FSM here, and I've been following the `module_process_event` for all the modules that I've been using. So while this is just a wrapper, I think it's good to stay as is for consistency.